### PR TITLE
Formalise method to layout subviews and constraints

### PR DESCRIPTION
**Pull request by marijnschilling**
#74

### CODE REVIEW

File: BEST_PRACTICES.md [hunk 1]
```diff
@@ -322,6 +322,40 @@ Views should be layout using Apple's Auto Layout. No third-party frameworks are
 
 Old style layout is still an option for when Auto Layout is not available.
 
+Use a method called `addSubViewsAndConstraints()` where you first add the sub views and then set your constraints. Call this method from `viewDidLoad()` in a ViewController or the `init(frame: CGRect)` in a View. We do this to keep the layout code from cluttering your init methods. We move adding the sub views in this method because there inseparably related (a.k.a: your app chrashes when setting constraints on a sub view that has not been added to the view.) It also avoids conflicts when you use subclassing and want to override the method where you set the constraints.
```
Comment (elland) [inline BEST_PRACTICES.md]: Shouldn't it be `addSubviewsAndConstraints`? Lowercase v for views? 👍

Comment (elland) [reply BEST_PRACTICES.md]: > We move adding the sub views in this method because there inseparably related

This needs some rewriting 😄

Comment (elland) [reply BEST_PRACTICES.md]: Also quite sure it's `subview` not `sub view`. 👍 We should probably mark the function as `open` in the example to make it clear that this is where you override layout, as well. 😄

Comment (marijnschilling) [reply BEST_PRACTICES.md]: Totally!

**Comment by marijnschilling** [comment]
🚣 ?

**Comment by 3lvis** [comment]
Great job @marijnschilling and @Elland. 👯

**Pull request merged**